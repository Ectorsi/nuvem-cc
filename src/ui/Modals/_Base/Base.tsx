import { CSSProperties, ReactNode } from 'react';
import ReactPortal from '../../../utils/reactPortals';
import * as S from './styles';

type ModalBaseProps = {
    children: ReactNode;
    showModal: boolean;
    handleShowModal: () => void;
    modalWrapperStyles?: CSSProperties;
};

export const Modal = ({
    children,
    handleShowModal,
    showModal,
}: ModalBaseProps) => {
    if (!showModal) {
        return null;
    }

    return (
        <ReactPortal wrapperId="react-portal-modal-container">
            <S.Overlay hidden={!showModal} />
            <S.Container>
                <S.CloseButton onClick={handleShowModal}>x</S.CloseButton>
                <S.Content>{children}</S.Content>
            </S.Container>
        </ReactPortal>
    );
};
