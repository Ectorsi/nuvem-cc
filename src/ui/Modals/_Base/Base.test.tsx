import React from "react";
import { render, screen } from "@testing-library/react";

import { Modal } from './Base';

const MockModalProps = {
    children: <div>Test</div>,
    showModal: true,
    handleShowModal: jest.fn(),
    modalWrapperStyles: {},
}

describe('PageTitle unit test', () => {
    it('should be able to mach snapshot', () => {
        const { container } = render(<Modal {...MockModalProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should be able dont to show the modal', () => {
        render(<Modal {...MockModalProps} />);
        const closeButton = screen.getByText(/x/i);
        expect(closeButton).toBeInTheDocument();
    });
});