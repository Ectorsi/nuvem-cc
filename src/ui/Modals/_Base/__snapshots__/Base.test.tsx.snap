// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle unit test should be able to mach snapshot 1`] = `null`;
